types/src/index.d.ts

```diff
@@ -0,0 +1,116 @@
+export = Anvil;
+declare class Anvil {
+    /**
+     * Perform some handy/necessary things for a GraphQL file upload to make it work
+     * with this client and with our backend
+     *
+     * @param  {string|Buffer} pathOrStreamLikeThing - Either a string path to a file,
+     *   a Buffer, or a Stream-like thing that is compatible with form-data as an append.
+     * @param  {object} formDataAppendOptions - User can specify options to be passed to the form-data.append
+     *   call. This should be done if a stream-like thing is not one of the common types that
+     *   form-data can figure out on its own.
+     *
+     * @return {UploadWithOptions} - A class that wraps the stream-like-thing and any options
+     *   up together nicely in a way that we can also tell that it was us who did it.
+     */
+    static prepareGraphQLFile(pathOrStreamLikeThing: string | Buffer, formDataAppendOptions: object): UploadWithOptions;
+    constructor(options: any);
+    options: any;
+    authHeader: string;
+    hasSetLimiterFromResponse: boolean;
+    limiterSettingInProgress: boolean;
+    rateLimiterSetupPromise: Promise<any>;
+    rateLimiterPromiseResolver: (value: any) => void;
+    _setRateLimiter({ tokens, intervalMs }: {
+        tokens: any;
+        intervalMs: any;
```

Comment:
Any easy way to inject types for simple things like this? i.e. this should be an int

Reply:
Yeah looks like it should generate properly with JSDoc comments. Will add that.